x-pack/plugins/index_management/public/application/components/component_templates/component_template_list/table.tsx

```diff
@@ -0,0 +1,224 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import React, { FunctionComponent, useState } from 'react';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n/react';
+import {
+  EuiInMemoryTable,
+  EuiButton,
+  EuiInMemoryTableProps,
+  EuiTableFieldDataColumnType,
+  EuiIcon,
+} from '@elastic/eui';
+
+import { EuiFlexGroup, EuiFlexItem } from '@elastic/eui';
+import { ComponentTemplateListItem } from '../types';
+
+export interface Props {
+  componentTemplates: ComponentTemplateListItem[];
+  onReloadClick: () => void;
+  onDeleteClick: (componentTemplateName: string[]) => void;
+}
+
+export const ComponentTable: FunctionComponent<Props> = ({
+  componentTemplates,
+  onReloadClick,
+  onDeleteClick,
+}) => {
+  const [selection, setSelection] = useState<ComponentTemplateListItem[]>([]);
+
+  const tableProps: EuiInMemoryTableProps<ComponentTemplateListItem> = {
+    itemId: 'name',
+    isSelectable: true,
+    'data-test-subj': 'componentTemplatesTable',
+    sorting: { sort: { field: 'name', direction: 'asc' } },
+    selection: {
+      onSelectionChange: setSelection,
+      selectable: ({ usedBy }) => usedBy.length === 0,
+      selectableMessage: (selectable) =>
+        !selectable
```

Comment:
nit: I prefer to use the affirmative in ternary

Reply:
Agreed